File: free-platform-framework/src/main/kotlin/top/yangwulang/platform/jimmer/interceptor/DataTypeBaseNoStatusInterceptor.java
package top.yangwulang.platform.jimmer.interceptor;

import cn.dev33.satoken.stp.StpUtil;
import com.alicp.jetcache.Cache;
import lombok.extern.slf4j.Slf4j;
import org.babyfish.jimmer.ImmutableObjects;
import org.babyfish.jimmer.sql.DraftInterceptor;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;
import top.yangwulang.platform.entity.DataTypeBaseNoStatusDraft;
import top.yangwulang.platform.entity.DataTypeBaseProps;
import top.yangwulang.platform.entity.sys.User;
import top.yangwulang.platform.utils.UserUtils;

import java.time.LocalDateTime;
import java.util.Date;

@Slf4j
@Component
public class DataTypeBaseNoStatusInterceptor implements DraftInterceptor<DataTypeBaseNoStatusDraft> {
    @Override
    public void beforeSave(@NotNull DataTypeBaseNoStatusDraft draft, boolean isNew) {
        try (Cache<String, Object> cache = UserUtils.loginUserCache()) {
            String loginId = (String) StpUtil.getTokenInfo().getLoginId();
            User user = (User) cache.get(loginId);
            if (!ImmutableObjects.isLoaded(draft, DataTypeBaseProps.UPDATE_DATE)) {
                draft.setUpdateDate(new Date());
                draft.setUpdateBy(user.userCode());
            }
            if (isNew) {
                draft.setCreateDate(new Date());
                draft.setCreateBy(user.userCode());
                draft.setUpdateBy(user.userCode());
                draft.setUpdateDate(new Date());
            } else {
                draft.setUpdateDate(new Date());
                draft.setUpdateBy(user.userCode());
            }
        } catch (Exception e) {
            log.error("全局拦截createBy异常", e);
            if (!ImmutableObjects.isLoaded(draft, DataTypeBaseProps.UPDATE_DATE)) {
                draft.setUpdateDate(new Date());
                draft.setUpdateBy("system");
            }
            if (isNew) {
                draft.setUpdateDate(new Date());
                draft.setCreateDate(new Date());
                draft.setCreateBy("system");
                draft.setUpdateBy("system");
            }
        }
    }
}

File: free-platform-framework/src/main/kotlin/top/yangwulang/platform/jimmer/interceptor/DataTypeBaseNoStatusInterceptor.kt
package top.yangwulang.platform.jimmer.interceptor

import cn.dev33.satoken.stp.StpUtil
import lombok.extern.slf4j.Slf4j
import org.babyfish.jimmer.ImmutableObjects
import org.babyfish.jimmer.sql.DraftInterceptor
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import top.yangwulang.platform.entity.DataTypeBaseNoStatus
import top.yangwulang.platform.entity.DataTypeBaseNoStatusDraft
import top.yangwulang.platform.entity.DataTypeBaseProps
import top.yangwulang.platform.entity.sys.User
import top.yangwulang.platform.utils.UserUtils
import java.util.*

@Slf4j
@Component
class DataTypeBaseNoStatusInterceptor : DraftInterceptor<DataTypeBaseNoStatus, DataTypeBaseNoStatusDraft> {
    private val logger: Logger = LoggerFactory.getLogger(DataTypeBaseNoStatusInterceptor::class.java)

    override fun beforeSave(draft: DataTypeBaseNoStatusDraft, original: DataTypeBaseNoStatus?) {
        try {
            UserUtils.loginUserCache().use { cache ->
                val loginId = StpUtil.getTokenInfo().getLoginId() as String
                val user = cache[loginId] as User
                if (!ImmutableObjects.isLoaded(draft, DataTypeBaseProps.UPDATE_DATE)) {
                    draft.setUpdateDate(Date())
                    draft.setUpdateBy(user.userCode())
                }
                if (original == null) {
                    draft.setCreateDate(Date())
                    draft.setCreateBy(user.userCode())
                    draft.setUpdateBy(user.userCode())
                    draft.setUpdateDate(Date())
                } else {
                    draft.setUpdateDate(Date())
                    draft.setUpdateBy(user.userCode())
                }
            }
        } catch (e: Exception) {
            logger.error("全局拦截createBy异常", e)
            if (!ImmutableObjects.isLoaded(draft, DataTypeBaseProps.UPDATE_DATE)) {
                draft.setUpdateDate(Date())
                draft.setUpdateBy("system")
            }
            if (original == null) {
                draft.setUpdateDate(Date())
                draft.setCreateDate(Date())
                draft.setCreateBy("system")
                draft.setUpdateBy("system")
            }
        }
    }
}
